sdk/internal/testframework/README.md

```diff
@@ -0,0 +1,161 @@
+# Azure SDK for Go Recorded Test Framework
+
+[![Build Status](https://dev.azure.com/azure-sdk/public/_apis/build/status/go/Azure.azure-sdk-for-go?branchName=master)](https://dev.azure.com/azure-sdk/public/_build/latest?definitionId=1842&branchName=master)
+
+The `testframework` package makes it easy to add recorded tests to your track-2 client package.
+Below are some examples that walk through setting up a recorded test end to end.
+
+## Examples
+
+### Initializing a Recording instance for a test
+
+The first step in instrumenting a client to interact with recorded tests is to create a `TestContext`.
+This acts as the interface between the recorded test framework and your chosen test package.
+In these examples we'll use testify's [assert](https://pkg.go.dev/github.com/stretchr/testify/assert),
+but you can use the framework of your choice.
+
+In the snippet below, demonstrates an example test setup func in which we are initializing the `TestContext`
+with the methods that will be invoked when your recorded test needs to Log, Fail, get the Name of the test,
+or indicate that the test IsFailed.
+
+***Note**: an instance of TestContext should be initialized for each test.*
+
+```go
+// a map to store our created test contexts
+var clientsMap map[string]*testContext = make(map[string]*testContext)
```

Comment:
QuickQuestion: This `testContext` will not be visible outside of the `testframework` package. Isn't this supposed to be a public one?

Reply:
yes, it should be. In fact I'm not sure why there is an unexported one. I think I left it there by accident.

Reply:
It's starting with small case so I don't think it'll be visible outside of `testframework` package.

Reply:
correct - we should only be referring to `testframework.TestContext` The local `testContext` is something else, which is just confusing because of the shared name.

Reply:
Now it all makes sense. Thanks 😄